jsk_vil_ros/node_scripts/ofa.py

```diff
@@ -0,0 +1,211 @@
+
+    def caption_action_cb(self, goal):
+        success = True
+        feedback = VQATaskFeedback()
+        result = VQATaskResult()
+        success = True
+        if goal.image.data:
+            result.image = goal.image
+        else:
+            rospy.loginfo("No images in goal message, so using subscribed image topic instead")
+            result.image = self.default_caption_img
+        img_byte = ros_img_to_base(result.image)
+        # creqte request
+        req = json.dumps({"image": img_byte,
+                          "queries": goal.queries}).encode("utf-8")
+        try:
+            response = self.send_request("caption", req)
+            if response.status_code == 200:
+                json_result = json.loads(response.text)["results"]
+                for res in json_result:
+                    msg = QuestionAndAnswerText()
+                    msg.question = res["question"]
+                    msg.answer = res["answer"]
+                    result.result.append(msg)
+            else:
+                rospy.logerr("Invalid http status code: {}".format(str(response.status_code)))
+                return
+        except Exception as e:
+            rospy.logerr(str(e))
+            feedback.status = str(e)
+            success = False
+        finally:
+            self.caption_as.publish_feedback(feedback)
+            result.done = success
+            self.caption_as.set_succeeded(result)
+
+    def vqa_topic_cb(self, data):
+        self.default_vqa_img = data
+        if not self.vqa_queries:
+            return
+        img_byte = ros_img_to_base(data)
+        queries = self.vqa_queries.split(";")
+        req = json.dumps({"image": img_byte,
+                          "queries": queries}).encode("utf-8")
+        try:
+            response = self.send_request("vqa_gen", req)
+        except ConnectionError as e:
+            rospy.logwarn_once("Cannot establish the connection with API server. Is it running?")
+        else:
+            if response.status_code == 200:
+                json_result = json.loads(response.text)
+                msg = VQAResult()
+                msg.image = data
+                multi_vis = ""
+                for result in json_result["results"]:
+                    result_msg = QuestionAndAnswerText()
+                    result_msg.question = result["question"]
+                    result_msg.answer = result["answer"]
+                    msg.result.append(result_msg)
+                    multi_vis += "Q:{}\n A:{}\n".format(result["question"],
+                                                        result["answer"])
+                self.vqa_image_pub.publish(data)
+                self.vqa_result_pub.publish(msg)
+                self.vqa_multi_vis_pub.publish(multi_vis)
+            else:
+                rospy.logerr("Invalid http status code: {}".format(str(response.status_code)))
+
+    def vqa_action_cb(self, goal):
+        success = True
```

Comment:
the default value should be `False` to prevent bugs.
and store `True` when it succeeded